src/common/src/array/decimal_array.rs

```diff
@@ -166,6 +168,16 @@ impl ArrayBuilder for DecimalArrayBuilder {
             data: self.data,
         })
     }
+
+    fn take(&mut self) -> Result<DecimalArray> {
```

Comment:
I'd prefer to create a new array builder every time we need to build a new array. Having such function in array builder could be confusing for future developers.